internal/measurexlite/conn.go

```diff
@@ -69,6 +69,80 @@ func (c *connTrace) Write(b []byte) (int, error) {
 	return count, err
 }
 
+// MaybeUDPLikeClose is a convenience function for closing a conn only when such a conn isn't nil.
+func MaybeCloseUDPLikeConn(conn model.UDPLikeConn) (err error) {
+	if conn != nil {
+		err = conn.Close()
+	}
+	return
+}
+
+// WrapUDPLikeConn returns a wrapped conn that saves network events into this trace.
+func (tx *Trace) WrapUDPLikeConn(conn model.UDPLikeConn) model.UDPLikeConn {
+	return &udpLikeConnTrace{
+		UDPLikeConn: conn,
+		tx:          tx,
+	}
+}
+
+// udpLikeConnTrace is a trace-aware model.UDPLikeConn.
+type udpLikeConnTrace struct {
+	// Implementation note: it seems safe to use embedding here because net.Conn
+	// is an interface from the standard library that we don't control
```

Comment:
```suggestion
	// Implementation note: it seems ~safe to use embedding here because model.UDPLikeConn
	// contains fields deriving from how lucas-clemente/quic-go uses the standard library
```